app_flutter/lib/state/brooks.dart

```diff
@@ -0,0 +1,78 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+typedef BrookCallback<T> = void Function(T event);
+
+/// A source of events of type T.
+///
+/// This is a light-weight stream, hence the name of the class.
+///
+/// Listeners cannot be registered multiple times simultaneously.
+class Brook<T> {
```

Comment:
This is a clever name
![brook: a small stream](https://user-images.githubusercontent.com/2148558/77130186-2f14bd80-6a14-11ea-8c79-1c4973f4bb33.png)

Reply:
maybe too clever, but it's buried in some random tiny codebase, so... :-)

Reply:
It's good. I got a chuckle when I saw the definition and had to share.